tonic/src/metadata/map.rs

```diff
@@ -237,6 +249,29 @@ impl MetadataMap {
         self.headers
     }
 
+    /// Convert a MetadataMap into a HTTP HeaderMap
+    ///
+    /// The returned map excludes headers reserved by the gRPC protocol.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use tonic::metadata::*;
+    /// let mut map = MetadataMap::new();
+    /// map.insert("grpc-timeout", "1S".parse().unwrap());
+    /// map.insert("x-host", "example.com".parse().unwrap());
+    ///
+    /// let http_map = map.into_sanitized_headers();
+    ///
+    /// assert!(http_map.get("grpc-timeout").is_none());
+    /// ```
+    pub fn into_sanitized_headers(mut self) -> http::HeaderMap {
```

Comment:
I think we should probably make this `pub(crate)` for now? Like we do with into http in request/response.